Testing ElevatedButton

Gustavoz9 opened the pull request:
resumo: 
testei o componente ButtonForTest encontrando o texto, a Key e o tipo dele
também testei se ele recebe clicks ( o apertar do botão )

ps... o main ocorre erro por causa do setState e não sei bem o motivo desse erro, mas falei com o Carlos ele falou que não precisava disso já que podia testar ele direto sem colocar na main ou precisar do setState.

exceção do erro:

- Exception has occurred.

 FlutterError (setState() or markNeedsBuild() called during build.
 This MyHomePage widget cannot be marked as needing to build because the framework is already in the process of building 
 widgets. A widget can be marked as needing to be built during the build phase only if one of its ancestors is currently building. 
 This exception is allowed because the framework builds parent widgets before children, which means a dirty descendant will 
 always be built. Otherwise, the framework might not visit this widget during this build phase.
 
 The widget on which setState() or markNeedsBuild() was called was:
-  MyHomePage

The widget which was currently being built when the offending call was made was:
-  ButtonForTest)

GustavoAFernnandes committed:
Testing ElevatedButton

carlosTsilva commented on lib/component/Button.dart:
```diff
@@ -0,0 +1,37 @@
+import 'package:flutter/src/foundation/key.dart';
+import 'package:flutter/src/widgets/framework.dart';
+
+class ButtonForTest extends StatefulWidget {
+  final Function click;
+  final Color cor;
+  final Color disableColor;
+  final String label;
+  final bool value;
+  const ButtonForTest(
+      {Key? key,
+      required this.click,
+      required this.cor,
+      required this.label,
+      required this.value,
+      required this.disableColor})
+      : super(key: key);
+
+  @override
+  State<ButtonForTest> createState() => _ButtonForTestState();
+}
+
+class _ButtonForTestState extends State<ButtonForTest> {
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton(
+      key: Key('findElevatedButton'),
+      onPressed: widget.click(),
+      child: Text(widget.label),
+      style: ButtonStyle(
+          backgroundColor: widget.value == true
+              ? MaterialStateProperty.all(widget.cor)
+              : MaterialStateProperty.all(widget.disableColor)),
+    );
+  }
```
Ele poderia ser um StatelesWidget

Gustavoz9 replied:
Mudei para o StatelessWidget

carlosTsilva commented on lib/component/Button.dart:
```diff
@@ -0,0 +1,37 @@
+      onPressed: widget.click(),
+      child: Text(widget.label),
+      style: ButtonStyle(
+          backgroundColor: widget.value == true
```
Nao precisa verificar se ele é igual a true da pra usar  widget.value ?

Gustavoz9 replied:
Sim acabei de mudar isso

carlosTsilva commented on test/widget_test.dart:
```diff
@@ -1,23 +1,32 @@
 void main() {
-  testWidgets(
-      'When the Helper has a title, it should render a Helper component',
+  testWidgets('must call back a functional widget',
```
A descrição do teste nao fica tao claro o q esta testando ..., se é um botão ou algo.
Link sobre nome de testes: https://www.notion.so/warrenbrasil/Testes-3799f27be42945c49035d9b892b18d40#c537574565c64b9194d3248e49dd23a3

Gustavoz9 replied:
Mudei para "WHEN button click THEN return Value" 
escolhi esse formato com base no exemplo do notion

Gustavoz9 committed:
Fix Change Description

mudei a descrição para algo mais adequado e claro, usando um formato base do exemplo desse notion:
https://www.notion.so/warrenbrasil/Testes-3799f27be42945c49035d9b892b18d40#c537574565c64b9194d3248e49dd23a3

Gustavoz9 committed:
Update Button.dart

Mudei o component para Statelees e apaguei a verificação do value na linha 27 no qual o tornava redundante.

Gustavoz9 committed:
Merge branch 'main' into ElevatedButton-Test
